mapbox-gl-style-spec package fails when bundled with Browserify

**mapbox-gl-js version**: 0.42

### Steps to Trigger Behavior

 1. Create a test directory and install `browserify` and `@mapbox/mapbox-gl-style-spec`.
 2. Add an `index.js` file with:

```js
var styleSpec = require('@mapbox/mapbox-gl-style-spec');
console.log(styleSpec.format({
  layers: []
}));
```

 3. Add an `index.html` file with:

```html
<!DOCTYPE html>
<html lang="en">
<body>
  <script src="./build.js"></script>
</body>
</html>
```

4. Run `npx browserify index.js > build.js`.
5. Open `index.html` in Chrome (where we wouldn't have to worry about any ES2015 business).

### Expected Behavior

Everything goes fine.

### Actual Behavior

Something goes wrong with module resolution:

```
Uncaught Error: Cannot find module 'sort-object'
    at s (_prelude.js:1)
    at _prelude.js:1
    at s (index.js:26)
    at index.js:28
    at Object.r.31../reference/latest.js (index.js:2989)
    at s (index.js:27)
    at index.js:28
    at Object.r.38../composite (index.js:7536)
    at s (index.js:27)
    at e (index.js:32)
```

If I comment out the lines that use `sort-object`, I get the same error for `lodash.isequal`, another external dependency.

--- a/src/style-spec/package.json
+++ b/src/style-spec/package.json
@@ -12,7 +12,7 @@
   "main": "./dist/index.js",
   "scripts": {
     "copy-flow-typed": "cp -R ../../flow-typed .",
-    "build": "../../node_modules/.bin/browserify style-spec.js --no-builtins --no-bundle-external --insert-global-vars=\"__filename,__dirname\" --standalone=mapboxGlStyleSpecification | ../../node_modules/.bin/derequire > dist/index.js",
+    "build": "../../node_modules/.bin/browserify style-spec.js --no-builtins --insert-global-vars=\"__filename,__dirname\" --standalone=mapboxGlStyleSpecification | ../../node_modules/.bin/derequire > dist/index.js",
     "prepublish": "in-publish && yarn copy-flow-typed && yarn build || not-in-publish",
     "postpublish": "in-publish && rm -r flow-typed dist/index.js || not-in-publish"
   },

